cmd/juju/commands/debuglog.go

```diff
@@ -213,36 +213,49 @@ func (c *debugLogCommand) processEntities(isCAAS bool, entities []string) []stri
 	if entities == nil {
 		return nil
 	}
-	result := make([]string, len(entities))
+	results := make([]string, len(entities))
 	for i, entity := range entities {
-		// A stringified unit or machine tag never match their "IsValid"
+		// Filter out valid tags early
+		if tag, err := names.ParseTag(entity); err == nil {
+			if tag.Kind() == names.UnitTagKind || tag.Kind() == names.MachineTagKind || tag.Kind() == names.ApplicationTagKind {
+				results[i] = entity
+				continue
+			}
+			logger.Warningf("Tag %s of kind %s is not a supported filter for debug-log", entity, tag.Kind())
+		}
+		// A stringified unit or machine tag are not themselves valid tags
 		// function from names, so if the string value passed in is a valid
 		// machine or unit, then convert here.
 		if names.IsValidMachine(entity) {
 			entity = names.NewMachineTag(entity).String()
 		} else if names.IsValidUnit(entity) {
 			entity = names.NewUnitTag(entity).String()
```

Comment:
I believe these lines can be removed. names.ParseTag calls IsValidMachine and IsValidUnit while parsing the tag. I'm wondering if this entire section can be simplified.

Not sure how `names.UnitTagKind + "-" + entity + "-*"` is handled.

Reply:
I don't think so unfortunately, before `ParseTag` calls `IsValidX` it processes the tag (splits on the first `-`, etc.

We will only get to this step if a complete tag has not been provided such as `0`, `ubuntu`, `ubuntu/0` which return error on ParseTag. However, `0` will return true for `names.IsValidMachine`, which then needs to be parsed into a tag string